samples/AzureStorageQueuesPubSubSample/AzureStorageQueuesPubSub.cs

```diff
@@ -0,0 +1,141 @@
+using Azure.Storage.Queues;
+using Dapr.PluggableComponents.Components;
+using Dapr.PluggableComponents.Components.PubSub;
+
+internal sealed class AzureStorageQueuesPubSub : IPubSub
+{
+    private readonly ILogger<AzureStorageQueuesPubSub> logger;
+
+    private string? connectionString;
+    private string? queueName;
+    private TimeSpan pollInterval = TimeSpan.FromSeconds(5);
+    private int maxMessages = 5; 
+
+    public AzureStorageQueuesPubSub(ILogger<AzureStorageQueuesPubSub> logger)
+    {
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    #region IPubSub Members
+
+    public Task InitAsync(MetadataRequest request, CancellationToken cancellationToken = default)
+    {
+        this.connectionString = request.Properties["connectionString"];
+        this.queueName = request.Properties["queueName"];
+
+        if (request.Properties.TryGetValue("pollInterval", out var pollIntervalString))
+        {
+            this.pollInterval = TimeSpan.FromSeconds(Int32.Parse(pollIntervalString));
+        }
+
+        if (request.Properties.TryGetValue("maxMessages", out var maxMessagesString))
+        {
+            this.maxMessages = Int32.Parse(maxMessagesString);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    public async Task PublishAsync(PubSubPublishRequest request, CancellationToken cancellationToken = default)
+    {
+        var queueClient = new QueueClient(this.connectionString, this.queueName);
+
+        await queueClient.CreateIfNotExistsAsync(cancellationToken: cancellationToken);
+
+        await queueClient.SendMessageAsync(Encoding.UTF8.GetString(request.Data.Span), cancellationToken);
+    }
+
+    public async Task PullMessagesAsync(PubSubPullMessagesTopic topic, IAsyncEnumerable<PubSubPullMessagesRequest> requests, IAsyncObserver<PubSubPullMessagesResponse> responses, CancellationToken cancellationToken = default)
+    {
+        var queueClient = new QueueClient(this.connectionString, this.queueName);
+
+        var acknowledgeTask =
+            async () =>
+            {
+                await foreach (var request in requests)
+                {
+                    if (request.AckMessageId != null)
+                    {
+                        string[] parsedMessageId = request.AckMessageId.Split("|");
+                        string messageId = parsedMessageId[0];
+                        string popReceipt = parsedMessageId[1];
+
+                        await queueClient.DeleteMessageAsync(messageId, popReceipt, cancellationToken);
```

Comment:
should we catch errors possibly thrown by the DeleteMessageAsync method?

Reply:
I guess the question for me would be whether such error handling overall enhances the sample (in terms of understanding what *should* be done for a production application) or detracts from the sample (in terms of clouding key aspects of the base implementation).  This is definitely on the "simpler is better" side, but I'm open to a more "best practices" style sample.  The difficulty with the latter is that it must stay be up to date with what's considered "best practice" over time as well as avoid giving users a false sense of security that this is the proper way to implement their specific (and varied) component (which may have wildly different concerns).

Reply:
That's a good question to be honest. At least from my perspective as a developer, I would like to see a simple but correct implementation (because error handling is also part of the sample otherwise it would cause the impression that errors don't need to be handled or doesn't happen at all?), the best of the worlds would be correctness and simplicity. That's why I think the samples are generally in-memory.

I guess for Azure Service Bus is really difficult to get the best of the worlds and It would work better if it is used by any [quickstart](https://github.com/dapr/quickstarts/) sample to understand the boundaries of the sample, how that's sounds?

TBH: I think this should be the way to go for all the sdks. Provide a component that work with an existing sample.

Thoughts @tmacam ?

Reply:
I've tended to view samples alongside product code in repos to be more of the "getting started" variety, with separate repos used for tutorials or "best practise", as those tend to have a lifetime and set of concerns independent of the SDK itself.

I think my concern would be that "correctness" is domain specific.  That is, the correct thing to do if one cannot delete the message from the queue after successfully being processed by the application, for example, may only be known to the user of the component and their specific circumstances. To be honest, I'd rather the pluggable component "blow up" than have logic that might otherwise paper over an error that is then borrowed for an implementation where that is no longer the correct thing to do.